core/kazoo_voicemail/src/kz_vm_message.erl

```diff
@@ -475,41 +587,70 @@ move_to_modb(AccountId, DocId, JObj, Funs) ->
                      | Funs
                     ],
     Options = [{'transform', fun(_, B) -> lists:foldl(fun(F, J) -> F(J) end, B, TransformFuns) end}],
+    try_move_to_modb(FromDb, FromId, ToDb, ToId, Options).
+
+-spec try_move_to_modb(ne_binary(), ne_binary(), ne_binary(), ne_binary(), kz_proplist()) -> db_ret().
+-spec try_move_to_modb(ne_binary(), ne_binary(), ne_binary(), ne_binary(), kz_proplist(), non_neg_integer()) -> db_ret().
+try_move_to_modb(FromDb, FromId, ToDb, ToId, Options) ->
+    try_move_to_modb(FromDb, FromId, ToDb, ToId, Options, 3).
+
+try_move_to_modb(_, _, ToDb, ToId, _, 0) ->
+        _ = kz_datamgr:del_doc(ToDb, ToId),
+        {'error', 'conflict'};
+try_move_to_modb(FromDb, FromId, ToDb, ToId, Options, Tries) ->
     case kz_datamgr:move_doc(FromDb, FromId, ToDb, ToId, Options) of
-        {'ok', _}=M ->
-            BoxId = kzd_box_message:source_id(JObj),
-            _ = update_mailbox(AccountId, BoxId, FromId),
-            _ = kz_datamgr:del_doc(get_db(AccountId), FromId),
-            M;
-        {'error', _}=E ->
-            lager:warning("failed to move voice message ~s to modb", [DocId]),
-            E
+        {'error', 'conflict'} ->
+            try_move_to_modb(FromDb, FromId, ToDb, ToId, Options, Tries - 1);
+        Other -> Other
     end.
 
--spec update_mailbox(ne_binary(), ne_binary(), ne_binary()) -> db_ret().
-update_mailbox(AccountId, BoxId, OldId) ->
+-spec update_media_id(ne_binary(), kz_json:object()) -> kz_json:object().
+update_media_id(MediaId, JObj) ->
+    Metadata = kzd_box_message:set_media_id(MediaId, kzd_box_message:metadata(JObj)),
+    kzd_box_message:set_metadata(Metadata, JObj).
+
+-spec cleanup_moved_msgs(ne_binary(), modb_move_ret()) -> 'ok'.
+cleanup_moved_msgs(_, []) -> 'ok';
+cleanup_moved_msgs(AccountId, [{_, J1}|_]=Moved) ->
+    lists:foreach(fun({OldId, NJObj}) ->
+                      NewId = kzd_box_message:media_id(NJObj),
+                      case kz_datamgr:del_doc(get_db(AccountId), OldId) of
+                          {'ok', _} -> 'ok';
+                          {'error', _R} ->
+                              lager:error("failed to remove voicemail message media doc "),
+                              lager:error("after moving to MODB(old_id: ~s new_id: ~s): ~p", [OldId, NewId, _R])
```

Comment:
Should maybe have only 1 call to log here: if datamgr calls fail then double the amount of messages will go to lager here.